files/build_templates/docker_image_ctl.j2

```diff
@@ -315,6 +315,35 @@ start() {
         source $PLATFORM_ENV_CONF
     fi
 
+    {%- if sonic_asic_platform == "broadcom" %}
+    {%- if docker_container_name == "syncd" %}
+    # Set the SYNCD_SHM_SIZE if this variable not defined
+    SYNCD_SHM_INI=/usr/share/sonic/device/profiles/syncd_shm.ini
```

Comment:
this syncd_shm.ini should be under x86_64-broadcom_common folder since it is brcm specific?

Reply:
Great. I am also considering this, and not sure which folder is better.
Yes, currently, this is only for Broadcom and can be placed in broadcom_common. 
But look at the folder "profiles", there just have some config for Broadcom as well, and syncd_shm.ini is a config for syncd container, and we specify the chip name in it, if we need add more for other vendor, we can add logic in this docker_image_ctl but not another profile?

Meanwhile we